examples/interactionsDemo.html

```diff
@@ -0,0 +1,526 @@
+                        $('#name').text('Name: ' + profileData.name);
+
+                        // Check if the current user will be able to even view the interaction before making the request
+                        let permissions = profileData.authorization.permissions;
+                        if (checkPermission(permissions, 'analytics:conversationDetail:view') &&
+                            (checkPermission(permissions, 'conversation:communication:view') ||
+                             checkPermission(permissions, 'quality:evaluation:add') ||
+                             checkPermission(permissions, 'quality:calibration:view') ||
+                             checkPermission(permissions, 'quality:evaluation:editScore'))) {
+                                canViewInteraction = true;
+                                $('#role').text('Role: Supervisor');
+                        } else {
+                                $('#role').text('Role: Agent');
+                        }
+
+                        if (checkPermission(permissions, 'routing:wrapupCode:view')) {
+                            canViewWrapUpCodes = true;
+                        }
+
+                           setHidden(authenticatingEl, true);
+                           makeSDKRequest();
+                    })
+                    .catch(function (err) {
+                        setHidden(authenticatingEl, true);
+                        if (!authenticated) {
+                            setErrorState('Failed to Authenticate with PureCloud');
+                        }
+                    });
+
+                /**
+                 * Makes the request to the API to get the current users interactions for
+                 * the past 30 days.
+                 */
+                function makeSDKRequest() {
+                    // 30 days in the past
+                    let today = new Date();
+                    let today30 = (new Date(new Date().setDate(today.getDate() - 30))).toISOString();
+
+                    var body = {
+                        "interval": today30 + "/" + today.toISOString(),
+                        "order": "asc",
+                        "orderBy": "conversationStart",
+                        "paging": {
+                            "pageSize": 25,
+                            "pageNumber": 1
+                        },
+                        "segmentFilters": [
+                            {
+                                "type": "and",
+                                "predicates": [
+                                    {
+                                        "type": "dimension",
+                                        "dimension": "userId",
+                                        "operator": "matches",
+                                        "value": this.profileData.id
+                                    }
+                                ]
+                            }
+                        ]
+                    };
+
```

Comment:
Do you need to disable and/or hide items here so it works in both the initial load and reload case?

Reply:
Can you clarify what you mean here? What items and what are you concerned might break because of them?

Reply:
iirc, I was thinking that you should "reset" the page state to the initial state here so you make sure it's in a correct state by the end of the function.  for example, maybe clear and hide the table and error message.  Not sure off the top of my head if there are more.

Now, it's also possible that you could handle these cases below.  For instance, only clear the error state in the success handler and only clear and hide the table in the error case.  Although this opens you up to get in a bad state if the function fails fast or errors out for some reason now or in the future.

Basically advocating for a "functional" style ui update here.  With that caveat that it's not functional at all since it has side effects ;)

Of course, if all of the cases are handled, success -> failure, failure -> success, failure -> failure, success -> success, I don't really mind how you get there.